x/oracle/simulations/operations.go

```diff
@@ -0,0 +1,321 @@
+		},
+	)
+
+	appParams.GetOrGenerate(cdc, OpWeightMsgDelegateFeedConsent, &weightMsgDelegateFeedConsent, nil,
+		func(_ *rand.Rand) {
+			weightMsgDelegateFeedConsent = simappparams.DefaultWeightMsgSetWithdrawAddress
+		},
+	)
+
+	return simulation.WeightedOperations{
+		simulation.NewWeightedOperation(
+			weightMsgAggregateExchangeRatePrevote,
+			SimulateMsgAggregateExchangeRatePrevote(ak, bk, k),
+		),
+		simulation.NewWeightedOperation(
+			weightMsgAggregateExchangeRateVote,
+			SimulateMsgAggregateExchangeRateVote(ak, bk, k),
+		),
+		simulation.NewWeightedOperation(
+			weightMsgDelegateFeedConsent,
+			SimulateMsgDelegateFeedConsent(ak, bk, k),
+		),
+	}
+}
+
+// SimulateMsgAggregateExchangeRatePrevote generates a MsgAggregateExchangeRatePrevote with random values.
+func SimulateMsgAggregateExchangeRatePrevote(
+	ak types.AccountKeeper,
+	bk types.BankKeeper,
+	k keeper.Keeper,
+) simtypes.Operation {
+	return func(
+		r *rand.Rand, app *baseapp.BaseApp, ctx sdk.Context, accs []simtypes.Account, chainID string,
+	) (simtypes.OperationMsg, []simtypes.FutureOperation, error) {
+		simAccount, _ := simtypes.RandomAcc(r, accs)
+		address := sdk.ValAddress(simAccount.Address)
+		// ensure the validator exists
+		val := k.StakingKeeper.Validator(ctx, address)
+		if val == nil || !val.IsBonded() {
+			return simtypes.NoOpMsg(
+				types.ModuleName,
+				types.TypeMsgAggregateExchangeRatePrevote,
+				"unable to find validator",
+			), nil, nil
+		}
+
+		prices := make(map[string]sdk.Dec, len(acceptList))
+		for _, denom := range acceptList {
+			prices[denom] = simtypes.RandomDecAmount(r, sdk.NewDec(1)).Add(umeePrice)
+		}
+
+		exchangeRatesStr := GenerateExchangeRatesString(prices)
+		voteHash := types.GetAggregateVoteHash(salt, exchangeRatesStr, address)
+
+		feederAddr, _ := k.GetFeederDelegation(ctx, address)
+		feederSimAccount, _ := simtypes.FindAccount(accs, feederAddr)
+
+		feederAccount := ak.GetAccount(ctx, feederAddr)
+		spendable := bk.SpendableCoins(ctx, feederAccount.GetAddress())
+
+		fees, err := simtypes.RandomFees(r, ctx, spendable)
+		if err != nil {
+			return simtypes.NoOpMsg(
+				types.ModuleName,
+				types.TypeMsgAggregateExchangeRatePrevote,
+				"unable to generate fees",
+			), nil, err
+		}
+
+		msg := types.NewMsgAggregateExchangeRatePrevote(voteHash, feederAddr, address)
+
+		txGen := simappparams.MakeTestEncodingConfig().TxConfig
+		tx, err := helpers.GenTx(
+			txGen,
+			[]sdk.Msg{msg},
+			fees,
+			helpers.DefaultGenTxGas,
+			chainID,
+			[]uint64{feederAccount.GetAccountNumber()},
+			[]uint64{feederAccount.GetSequence()},
+			feederSimAccount.PrivKey,
+		)
+		if err != nil {
+			return simtypes.NoOpMsg(types.ModuleName, msg.Type(), "unable to generate mock tx"), nil, err
+		}
+
+		_, _, err = app.Deliver(txGen.TxEncoder(), tx)
+		if err != nil {
+			return simtypes.NoOpMsg(types.ModuleName, msg.Type(), "unable to deliver tx"), nil, err
+		}
+
+		voteHashMap[address.String()] = exchangeRatesStr
```

Comment:
I see why this `voteHashMap` is needed, but are we sure it's safe for use between simulation iterations?

Reply:
We can definitely put this in an instance of a struct or something

Reply:
Localized it to `WeightedOperations`, let me know if that's alright

Reply:
I think that should work 👍